lib/spack/spack/solver/concretize.lp

```diff
@@ -307,29 +289,25 @@ real_node(Package) :- node(Package), not external(Package).
 
 % if an external version is selected, the package is external and
 % we are using the corresponding spec
+% TODO: this only checks that the *version* is the same, but not that
+% TODO: other attributes are. Consider modeling externals with imposed
+% TODO: constraints (to ensure they all match), not by writing out the
+% TODO: match criteria like this (we'll always forget something)
```

Comment:
This is legacy code stemming from the previous formulation of rules for externals. Everything is checked, but it's indirect and definitely not clear (we'll use the external version weight for external `ID` only if external `ID` is the one selected). I'll check if we can write this rule in a way it's less cryptic.

Reply:
This should be a better substitute for the rule below the comment:
```
% a package is external if we are using an external spec for it
external(Package) :- external_spec_selected(Package, _).

% we can't use the weight for an external version if we don't use the 
% corresponding external spec.
:- version(Package, Version),
   version_weight(Package, Weight),
   external_version_declared(Package, Version, Weight, ID),
   not external(Package).
```
(or at least it seems clearer to me).

Reply:
ok this is addressed.